json-decode.go

```diff
@@ -100,16 +97,27 @@ func (cdc *Codec) decodeReflectJSON(bz []byte, info *TypeInfo, rv reflect.Value,
 	//----------------------------------------
 	// Signed, Unsigned
 
-	case reflect.Int64, reflect.Int32, reflect.Int16, reflect.Int8, reflect.Int,
-		reflect.Uint64, reflect.Uint32, reflect.Uint16, reflect.Uint8, reflect.Uint:
+	case reflect.Int64, reflect.Int:
+		fallthrough
```

Comment:
Do you think splitting up the cases (Int64, Int &  Uint64, Uint) increases readability here? It confused me on first sight. But it is actually straight forward.